lib/stamp/emitters/composite.rb

```diff
@@ -7,12 +7,9 @@ def initialize
         @emitters = []
       end
 
-      def format(target)
-        # NOTE using #each to build string because benchmarking shows
-        # that it's ~20% faster than .map.join('')
-        result = ''
-        @emitters.each { |e| result << e.format(target).to_s }
-        result
+      def format(out, target)
```

Comment:
yes. that is all it is.
In the cases where you want to output more than 1 string, it ends up being quicker to `lambda |result, target| result << str1 << str2 }` than `result << lambda { |target| str1 + str2 }`.

Granted it is probably 10% in trivial benchmarks, but hey. that is something right?
(well, if you think it is much uglier, than that probably doesn't warrant the change)

It is hard to benchmark across these changes. I like benchmark-ips, but not sure how to do it across sweeping changes like this. Only can do it with lambdas.